reporter/internal/pdata/generate.go

```diff
@@ -29,16 +30,27 @@ const (
 // Generate generates a pdata request out of internal profiles data, to be
 // exported.
 func (p *Pdata) Generate(tree samples.TraceEventsTree,
-	agentName, agentVersion string) pprofile.Profiles {
+	agentName, agentVersion string) (pprofile.Profiles, error) {
 	profiles := pprofile.NewProfiles()
+	dic := profiles.ProfilesDictionary()
+
+	// Temporary helpers that will build the various tables in ProfilesDictionary.
+	stringSet := libpf.OrderedSet[string]{}
+	funcSet := libpf.OrderedSet[samples.FuncInfo]{}
+	mappingSet := libpf.OrderedSet[libpf.FileID]{}
```

Comment:
Maybe start with a reasonable size to avoid some of the reallocations when growing.

Reply:
I switched to `64`, if something more elaborate is needed, best done in a new PR.